core/src/main/java/com/linecorp/armeria/internal/common/SchemeAndAuthority.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2024 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.internal.common;
+
+import static java.util.Objects.requireNonNull;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+import com.google.common.base.Ascii;
+
+import com.linecorp.armeria.common.annotation.Nullable;
+
+public final class SchemeAndAuthority {
+    /**
+     * Validator for the scheme part of the URI, as defined in
+     * <a href="https://datatracker.ietf.org/doc/html/rfc3986#section-3.1">the section 3.1 of RFC3986</a>.
+     */
+    private static final Predicate<String> SCHEME_VALIDATOR =
+            scheme -> Pattern.compile("^([a-zA-Z][a-zA-Z0-9+\\-.]*)").matcher(scheme).matches();
```

Comment:
I see there are two `SCHEME_VALIDATOR`s in this pull request and they don't need to be a `Predicate`. Could we add a utility method `isValidScheme(String)` to `ArmeriaHttpUtil` to replace both `SCHEME_VALIDATOR`s?

Reply:
Yes! It looks good to make `ArmeriaHttpUtil.isValidScheme(String)`!